lib/trento_web/views/v1/health_overview_view.ex

```diff
@@ -37,12 +48,21 @@ defmodule TrentoWeb.V1.HealthOverviewView do
   defp extract_database_id([%DatabaseInstanceReadModel{sap_system_id: sap_system_id} | _]),
     do: sap_system_id
 
-  @spec extract_cluster_id([DatabaseInstanceReadModel.t()]) :: String.t()
+  @spec extract_cluster_id([ApplicationInstanceReadModel.t() | DatabaseInstanceReadModel.t()]) ::
+          String.t()
   defp extract_cluster_id([]), do: nil
 
   defp extract_cluster_id([%DatabaseInstanceReadModel{host: %{cluster_id: cluster_id}} | _]),
     do: cluster_id
 
+  defp extract_cluster_id(application_instances) do
+    Enum.find_value(application_instances, nil, fn
```

Comment:
Now that I see this I am wondering whether the previous still makes sense :thinking: 
```elixir
defp extract_cluster_id([%DatabaseInstanceReadModel{host: %{cluster_id: cluster_id}} | _]),
    do: cluster_id
```
We are basically getting whatever the cluster id is on the first instance of the list, while in the similar function in the service we are looking in all the instances to find a cluster id, if any.

I am bit confused :sweat_smile:

Reply:
Alright, thanks for the refresher :smile: